File: MyManagerCSharp/Log/LogUserManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MyManagerCSharp.Log
{
    public class LogUserManager : ManagerDB
    {

        public enum LogType
        {
            Login,
            LoginMobile,
            Logout,
            Reset_password,
            Update_password,
            Download,
            Not_authorized,
            Not_authenticated,
            Page_not_found,
            Account_enable,
            Account_disable,
            Account_delete,
            Exception,
            Error,
            Report,
            ControllerAction,
            Access_denied
        }


        public LogUserManager(string connectionName)
            : base(connectionName)
        {

        }

        public LogUserManager(System.Data.Common.DbConnection connection)
            : base(connection)
        {

        }

        public void insert(long userId, string login, LogType tipo)
        {
            insert(userId, login, tipo, "", null, "", "", null,"");
        }

        public void insert(long userId, string login, LogType tipo, System.Net.IPAddress ipAddress)
        {
            insert(userId, login, tipo, "", ipAddress, "", "", null, "");
        }

        public void insert(long userId, string login, LogType tipo, string nota)
        {
            insert(userId, login, tipo, nota, null, "", "", null, "");
        }

        public void insert(long userId, string login, LogType tipo, string nota, System.Net.IPAddress ipAddress, string controller, string action, Guid? sessionId,  string httpMethod)
        {
            string strSQLParametri;

            mStrSQL = "INSERT INTO MyLogUser ( DATE_ADDED , user_id , tipo";
            strSQLParametri = " VALUES ( GetDate() , " + userId + ", '" + tipo.ToString() + "' ";

            System.Data.Common.DbCommand command;
            command =  mConnection.CreateCommand();
            command.Connection =  mConnection;

            if (!String.IsNullOrEmpty(nota))
            {
                mStrSQL += ",nota ";
                strSQLParametri += ", @nota ";
                mAddParameter(command, "@nota", nota);
            }

            if (!String.IsNullOrEmpty(login))
            {
                mStrSQL += ",login ";
                strSQLParametri += ", @LOGIN ";
                mAddParameter(command, "@LOGIN", login);
            }

            if (!String.IsNullOrEmpty(controller))
            {
                mStrSQL += ",controller ";
                strSQLParametri += ", @CONTROLLER ";
                mAddParameter(command, "@CONTROLLER", controller);
            }

            if (!String.IsNullOrEmpty(action))
            {
                mStrSQL += ",action ";
                strSQLParametri += ", @ACTION ";
                mAddParameter(command, "@ACTION", action);
            }

            if (!String.IsNullOrEmpty(httpMethod))
            {
                mStrSQL += ",http_method ";
                strSQLParametri += ", @HTTP_METHOD ";
                mAddParameter(command, "@HTTP_METHOD", httpMethod);
            }
            

            if (ipAddress != null)
            {
                mStrSQL += ",ip_address ";
                strSQLParametri += ", @IP ";
                mAddParameter(command, "@IP", ipAddress.ToString());
            }

            if (sessionId != null)
            {
                mStrSQL += ",session_id ";
                strSQLParametri += ", @SESSION_ID ";
                mAddParameter(command, "@SESSION_ID", sessionId);
            }

            command.CommandText = mStrSQL + " ) " + strSQLParametri + " )";
            command.CommandType = System.Data.CommandType.Text;

            mExecuteNoQuery(command);
        }







    }
}

File: MyManager/LogUserManager.vb

''' <summary>
''' La Classe si occupa di loggare le azioni di un utente.
''' Ad esempio con LogType = Login logghiamo tutti gli accessi di un utente
''' </summary>
''' <remarks></remarks>

Public Class LogUserManager
    Inherits Manager


    Public Enum LogType
        Login
        ResetPassword
        UpdatePassword
        URL
        Download
        NotAuthorized
        NotAuthenticated
        PageNotFound
        AccountEnable
        AccountDisable
        AccountDelete
    End Enum


    Public Sub New()
        MyBase.New("utenti")
    End Sub

    Public Sub New(ByVal connectionName As String)
        MyBase.New(connectionName)
    End Sub

    Public Sub New(ByVal connection As System.Data.Common.DbConnection)
        MyBase.New(connection)
    End Sub

    Private Function getNewLogUserId() As Long
        Return Me._getSequence("S_LOG_UTENTE_ID")
    End Function

    Public Function insert(ByVal userId As Long, ByVal tipo As LogUserManager.LogType, ByVal nota As String) As Long
        Dim ctipo As String = tipo.ToString
        'id della login ricavata dalla sequence
        'Dim logUserId As Long = Me.getNewLogUserId
        Dim strSQL As String = ""


        'Per oracle
        'strSQL = "insert into log_utente (log_utente_id, user_id, type, nota, date_added) values " & _
        '               "(?, ?, ?, ?, sysdate)"

        'Per Access
        strSQL = "insert into log_utente (user_id, type, nota)  "
        Dim strSQLParametri As String = " values ("

        Dim command As System.Data.Common.DbCommand
        command = _connection.CreateCommand()
        If (userId <> -1) Then
            strSQLParametri &= userId
        Else
            strSQLParametri &= "null "
        End If



        If String.IsNullOrEmpty(tipo.ToString) Then
            strSQLParametri &= ", null "
        Else
            strSQLParametri &= ",@tipo "
            Me._addParameter(command, "@tipo", tipo.ToString)
        End If


        If String.IsNullOrEmpty(nota) Then
            strSQLParametri &= ", null "
        Else
            strSQLParametri &= ",@nota "
            Me._addParameter(command, "@nota", nota)
        End If


        command.CommandText = strSQL & strSQLParametri & " )"
        command.CommandType = CommandType.Text

        command.ExecuteNonQuery()

        Return 0
    End Function

    Public Function insert(ByVal userId As Long, ByVal tipo As LogUserManager.LogType) As Long
        Return insert(userId, tipo, Nothing)
    End Function


End Class
